--- ./include/sys/cdefs.h.orig	2019-06-17 21:52:28.792878124 +0000
+++ ./include/sys/cdefs.h	2019-06-17 21:53:17.297268601 +0000
@@ -38,6 +38,27 @@
 
 #define __BIONIC__ 1
 
+#ifndef __has_extension
+#define __has_extension         __has_feature
+#endif
+#ifndef __has_feature
+#define __has_feature(x)        0
+#endif
+#ifndef __has_include
+#define __has_include(x)        0
+#endif
+#ifndef __has_builtin
+#define __has_builtin(x)        0
+#endif
+#ifndef __has_attribute
+#define __has_attribute(x)      0
+#endif
+
+#if !defined(__clang__)
+#define _Nonnull
+#define _Nullable
+#endif
+
 #if defined(__cplusplus)
 #define __BEGIN_DECLS extern "C" {
 #define __END_DECLS }
@@ -296,6 +317,11 @@
 #  define __BIONIC_INCLUDE_FORTIFY_HEADERS 1
 #endif
 
+#if !defined(__clang__) && defined(__BIONIC_INCLUDE_FORTIFY_HEADERS)
+#undef __BIONIC_INCLUDE_FORTIFY_HEADERS
+#warning "fortify headers not supported and not used with gcc"
+#endif
+
 #define __overloadable __attribute__((overloadable))
 
 /* Used to tag non-static symbols that are private and never exposed by the shared library. */
